src/Auth/Models/User.php

```diff
@@ -154,17 +154,6 @@ public function afterLogin()
         $this->forceSave();
     }
 
-    /**
-     * Delete the user groups
-     * @return void
-     */
-    public function afterDelete()
-    {
-        if ($this->hasRelation('groups')) {
-            $this->groups()->detach();
-        }
-    }
```

Comment:
@mjauvin this won't have any ramifications with the User plugin, yeah? Since that also extends this class (and potentially other plugins do too).

Reply:
We may need to add 'delete' => true to its groups belongsToMany relation, you're right

Reply:
@bennothommo actually, this is fine as the pivot records for the relations now get deleted always as they should from the base Model class, unless of course the relation has been marked as "softDelete => true", in which case only the deleted_at column would get updated.

So the previous behavior is preserved, the only difference is its now done in the base Model's `performDeleteOnRelations()` method for all models instead of being "patched" in only for the Auth\User model.